sql/src/main/java/org/apache/apex/malhar/sql/planner/RelNodeVisitor.java

```diff
@@ -0,0 +1,457 @@
+  }
+
+  public RelInfo visitCollect(Collect collect, List<RelInfo> inputStreams) throws Exception
+  {
+    return defaultValue(collect, inputStreams);
+  }
+
+  public RelInfo visitCorrelate(Correlate correlate, List<RelInfo> inputStreams) throws Exception
+  {
+    return defaultValue(correlate, inputStreams);
+  }
+
+  public RelInfo visitDelta(Delta delta, List<RelInfo> inputStreams) throws Exception
+  {
+    ConsoleOutputOperator console = dag.addOperator(OperatorUtils.getUniqueOperatorName(delta.getRelTypeName()),
+        ConsoleOutputOperator.class);
+    console.setStringFormat("Delta Record: %s");
+
+    return new RelInfo("Delta", Lists.<Operator.InputPort>newArrayList(console.input), console, null,
+      delta.getRowType());
+  }
+
+  public RelInfo visitExchange(Exchange exchange, List<RelInfo> inputStreams) throws Exception
+  {
+    return defaultValue(exchange, inputStreams);
+  }
+
+  public RelInfo visitProject(Project project, List<RelInfo> inputStreams) throws Exception
+  {
+    if (inputStreams.size() == 0 || inputStreams.size() > 1) {
+      throw new UnsupportedOperationException("Project is a SingleRel");
+    }
+
+    FilterTransformOperator operator = dag.addOperator(OperatorUtils.getUniqueOperatorName(project.getRelTypeName()),
+        FilterTransformOperator.class);
+    Map<String, String> expMap = new HashMap<>();
+    ExpressionCompiler compiler = new ExpressionCompiler(new RexBuilder(project.getCluster().getTypeFactory()));
+
+    for (Pair<RelDataTypeField, RexNode> pair : Pair.zip(project.getRowType().getFieldList(), project.getProjects())) {
+      String fieldName = OperatorUtils.getFieldName(pair.left);
+      String expression = compiler.getExpression(pair.right, project.getInput().getRowType(), project.getRowType());
+      expMap.put(fieldName, expression);
+    }
+    operator.setExpressionMap(expMap);
+
+    return new RelInfo("Project", Lists.<Operator.InputPort>newArrayList(operator.input), operator, operator.output,
+      project.getRowType());
+  }
+
+  public RelInfo visitFilter(Filter filter, List<RelInfo> inputStreams) throws Exception
+  {
+    if (inputStreams.size() == 0 || inputStreams.size() > 1) {
+      throw new UnsupportedOperationException("Filter is a SingleRel");
+    }
+
+    FilterTransformOperator operator = dag.addOperator(OperatorUtils.getUniqueOperatorName(filter.getRelTypeName()),
+        FilterTransformOperator.class);
+    ExpressionCompiler compiler = new ExpressionCompiler(new RexBuilder(filter.getCluster().getTypeFactory()));
+    String expression = compiler.getExpression(filter.getCondition(), filter.getInput().getRowType(),
+        filter.getRowType());
+
+    Map<String, String> expMap = new HashMap<>();
+    for (Pair<RelDataTypeField, RelDataTypeField> pair : Pair.zip(filter.getInput().getRowType().getFieldList(),
+        filter.getRowType().getFieldList())) {
+      String leftName = OperatorUtils.getFieldName(pair.left);
```

Comment:
They're different as per design by Calcite. Not sure what should be documented here.